tests/xbot/common/controls/sensors/JoystickButtonManagerTest.java

```diff
@@ -16,25 +19,36 @@ public void testAvailability() {
 
         WPIFactory factory = this.injector.getInstance(WPIFactory.class);
         XJoystick testJoystick = factory.getJoystick(1);
+        RobotAssertionManager assertion = this.injector.getInstance(RobotAssertionManager.class);
 
-        JoystickButtonManager testButtons = new JoystickButtonManager(12, factory, testJoystick);
+        JoystickButtonManager testButtons = new JoystickButtonManager(12, factory, assertion, testJoystick);
 
         int i = 13;
-        assertTrue("Button " + i + " should be null.", null == testButtons.getifAvailable(i));
+        testButton(testButtons, i);
 
         i = 0;
-        assertTrue("Button " + i + " should be null.", null == testButtons.getifAvailable(i));
+        testButton(testButtons, i);
 
         i = -1;
-        assertTrue("Button " + i + " should be null.", null == testButtons.getifAvailable(i));
+        testButton(testButtons, i);
 
         for (int x = 1; x <= 12; x++) {
             assertTrue("Button " + x + " should not be null.", null != testButtons.getifAvailable(x));
         }
         for (int x = 1; x <= 12; x++) {
-            assertTrue("Button " + x + " should be null.", null == testButtons.getifAvailable(x));
+            testButton(testButtons, x);
         }
 
     }
+    
+    private void testButton(JoystickButtonManager manager, int i) {
```

Comment:
jUnit has an annotation for expected exceptions. I think it would be best to actually split this up into separate tests and set the exception that you want to see.